x-pack/legacy/plugins/siem/public/components/open_timeline/index.tsx

```diff
@@ -161,79 +164,81 @@ export const StatefulOpenTimelineComponent = React.memo<OpenTimelineOwnProps>(
       const { index, size } = page;
       const { field, direction } = sort;
       setPageIndex(index);
       setPageSize(size);
       setSortDirection(direction);
       setSortField(field);
-    };
+    }, []);
 
     /** Invoked when the user toggles the option to only view favorite timelines */
-    const onToggleOnlyFavorites: OnToggleOnlyFavorites = () => {
+    const onToggleOnlyFavorites: OnToggleOnlyFavorites = useCallback(() => {
       setOnlyFavorites(!onlyFavorites);
-    };
+    }, [onlyFavorites]);
 
     /** Invoked when the user toggles the expansion or collapse of inline notes in a table row */
-    const onToggleShowNotes: OnToggleShowNotes = (
-      newItemIdToExpandedNotesRowMap: Record<string, JSX.Element>
-    ) => {
-      setItemIdToExpandedNotesRowMap(newItemIdToExpandedNotesRowMap);
-    };
+    const onToggleShowNotes: OnToggleShowNotes = useCallback(
+      (newItemIdToExpandedNotesRowMap: Record<string, JSX.Element>) => {
+        setItemIdToExpandedNotesRowMap(newItemIdToExpandedNotesRowMap);
+      },
+      []
+    );
 
     /** Resets the selection state such that all timelines are unselected */
-    const resetSelectionState = () => {
+    const resetSelectionState = useCallback(() => {
       setSelectedItems([]);
-    };
+    }, []);
 
-    const openTimeline: OnOpenTimeline = ({
-      duplicate,
-      timelineId,
-    }: {
-      duplicate: boolean;
-      timelineId: string;
-    }) => {
-      if (isModal && closeModalTimeline != null) {
-        closeModalTimeline();
-      }
+    const openTimeline: OnOpenTimeline = useCallback(
+      ({ duplicate, timelineId }: { duplicate: boolean; timelineId: string }) => {
+        if (isModal && closeModalTimeline != null) {
+          closeModalTimeline();
+        }
 
-      queryTimelineById({
-        apolloClient,
-        duplicate,
-        timelineId,
-        updateIsLoading,
-        updateTimeline,
-      });
-    };
+        queryTimelineById({
+          apolloClient,
+          duplicate,
+          timelineId,
+          updateIsLoading,
+          updateTimeline,
+        });
+      },
+      [apolloClient, updateIsLoading, updateTimeline]
+    );
+
+    const deleteTimelines: DeleteTimelines = useCallback(
+      (timelineIds: string[], variables?: AllTimelinesVariables) => {
+        if (timelineIds.includes(timeline.savedObjectId || '')) {
+          createNewTimeline({ id: 'timeline-1', columns: defaultHeaders, show: false });
+        }
+        apolloClient.mutate<DeleteTimelineMutation.Mutation, DeleteTimelineMutation.Variables>({
+          mutation: deleteTimelineMutation,
+          fetchPolicy: 'no-cache',
+          variables: { id: timelineIds },
+          refetchQueries: [
+            {
+              query: allTimelinesQuery,
+              variables,
+            },
+          ],
+        });
+      },
+      [apolloClient, createNewTimeline]
```

Comment:
Necessary dependency?

```suggestion
      [apolloClient, createNewTimeline, timeline]
```